.github/workflows/compile_test.yml

```diff
@@ -0,0 +1,59 @@
+name: Compile Test
+on: [push, pull_request]
+jobs:
+  compiletest:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+    - name: Download Kernel Source
+      run: |
+        KVER=$(cat version)
+        _RC=$(echo $KVER | tr -d 1234567890.)
+        
+        if [ -n "$_RC" ]; then
+          KSRC_URL=https://git.kernel.org/torvalds/t/linux-${KVER}.tar.gz
+        else
+          KSRC_URL=https://cdn.kernel.org/pub/linux/kernel/v${KVER//.*}.x/linux-${KVER}.tar.xz
+        fi
+        
+        curl $KSRC_URL | tar -xJ
+    - name: Install Build Dependencies
+      run: |
+        sudo apt-get install -y build-essential fakeroot libncurses-dev bison flex libssl-dev libelf-dev \
+            openssl dkms libudev-dev libpci-dev libiberty-dev autoconf wget xz-utils git \
+            libcap-dev bc rsync cpio dh-modaliases debhelper kernel-wedge curl gawk dwarves zstd
+    - name: Checkpatch
+      run: |
+        KVER=$(cat version)
+        cd linux-$KVER
+        # Skip apple-bce because it needs a lot of formatting changes, and
```

Comment:
We probably can exclude bcm5974 and applesmc from checkpatch since they shall not be sent upstream